### PR TITLE
i#5269: Upgrade to clang-format-9

### LINKED ISSUE #5269: update to clang-format-9
We are currently using clang-format-6 but it is getting difficult to maintain that on new systems.  This is a proposal to update to clang-format-9.  Some files do have drastic differences with version 9, such as optionsx.h, so we'd do a single reformat across the code base.

### PR DESCRIPTION
**Pull request by derekbruening**
Updates the official clang-format version from 6.0 to 9.0.  Re-formats
all of the .h, .c, and .cpp files, which results in mostly minor
changes, but substantial changes in core/optionsx.h and
core/lib/globals_api.h.

Fixes #5269

### CODE REVIEW

File: clients/drcachesim/analysis_tool.h [hunk 1]
```diff
@@ -90,8 +90,8 @@ class analysis_tool_t {
      * error message.
```
Comment (abhinav92003) [inline clients/drcachesim/analysis_tool.h:90]: Update year range in Copyright notice.

File: clients/drcachesim/tests/heap_test.cpp [hunk 2]
```diff
@@ -71,36 +71,36 @@ test_operators()
             int x = 0;
```
Comment (abhinav92003) [inline clients/drcachesim/tests/heap_test.cpp:71]: Update year range in Copyright notice.

File: core/annotations.c [hunk 3]
```diff
@@ -352,12 +352,13 @@ instrument_annotation(dcontext_t *dcontext, IN OUT app_pc *start_pc,
 #    endif
```
Comment (abhinav92003) [inline core/annotations.c:352]: Update year range in Copyright notice, and elsewhere too.

**Review by abhinav92003 — APPROVED** [review]

File: clients/drcachesim/analysis_tool.h (hunk 1 above)
Comment (derekbruening) [reply clients/drcachesim/analysis_tool.h:90]: Done for all files changed here.

File: clients/drcachesim/tests/heap_test.cpp (hunk 2 above)
Comment (derekbruening) [reply clients/drcachesim/tests/heap_test.cpp:71]: Done for all files changed here.

File: core/annotations.c (hunk 3 above)
Comment (derekbruening) [reply core/annotations.c:352]: Done for all files changed here.

**Pull request merged**